python/graphstorm/trainer/gsgnn_trainer.py

```diff
@@ -186,12 +187,15 @@ def save_model(self, model, epoch, i, save_model_path):
         '''Save the model for a certain iteration in an epoch.
         '''
         th.distributed.barrier()
-        if save_model_path is not None and self.rank == 0:
+        if save_model_path is not None:
+            assert isinstance(model.module, GSgnnModel), \
+                "Please make sure the model derives from GSgnnModel, " \
+                "which provides a scalable model saving implementation."
             save_model_path = self._gen_model_path(save_model_path, epoch, i)
             model.module.save_model(save_model_path)
             self.optimizer.save_opt_state(save_model_path)
```

Comment:
do you need to change anything in optimizer to save the states of the learnable embeddings?

Reply:
We do not support loading the states of the learnable embeddings: https://github.com/awslabs/graphstorm/blob/main/python/graphstorm/model/utils.py#L336-L337. 
Let me add an issue about it. I will not support this in this PR.